TypeScript compilation error

I'm trying to use Papr in my TypeScript project, but I'm getting a compilation error:

```
node_modules/papr/esm/utils.d.ts(24,168): error TS2344: Type 'keyof TDefaults' does not satisfy the constraint 'keyof TSchema'.
  Type 'string | number | symbol' is not assignable to type 'keyof TSchema'.
    Type 'string' is not assignable to type 'keyof TSchema'.
```

Papr version: `2.0.2`
TypeScript version: `4.2.3`

TypeScript config:
```
{
	"include": ["./src"],
	"compilerOptions": {
		"target": "ESNext",
		"lib": ["ESNext"],
		"module": "CommonJS",
		"moduleResolution": "Node",
		"esModuleInterop": true,
		"rootDir": "./src",
		"outDir": "./build",
		"strict": true,
		"noUnusedLocals": true,
		"noUnusedParameters": true,
		"noImplicitReturns": true,
		"noFallthroughCasesInSwitch": true,
		"forceConsistentCasingInFileNames": true
	}
}

```

--- a/src/utils.ts
+++ b/src/utils.ts
@@ -27,9 +27,11 @@ export interface ModelOptions {
   maxTime?: number;
 }
 
-export type DocumentForInsertWithoutDefaults<TSchema, TDefaults extends Partial<TSchema>> =
-  // @ts-expect-error: This works, but TS complains that we can't use `keyof TDefaults` to index `TSchema`
-  Omit<OptionalId<TSchema>, keyof TDefaults> & Partial<Pick<TSchema, keyof TDefaults>>;
+export type DocumentForInsertWithoutDefaults<TSchema, TDefaults extends Partial<TSchema>> = Omit<
+  OptionalId<TSchema>,
+  keyof TDefaults
+> &
+  Partial<Pick<TSchema, keyof TDefaults & keyof TSchema>>;
 
 export type DocumentForInsert<TSchema, TDefaults extends Partial<TSchema>> = Extract<
   keyof TSchema,
@@ -44,8 +46,7 @@ export type ProjectionType<
   Projection extends Partial<Record<keyof TSchema, number>> | undefined
 > = undefined extends Projection
   ? TSchema
-  : // @ts-expect-error: This works, but TS complains that we can't use `keyof Projection` to index `TSchema`
-    BaseSchema & Pick<TSchema, keyof Projection>;
+  : BaseSchema & Pick<TSchema, keyof Projection & keyof TSchema>;
 
 export function getIds(ids: (string | ObjectId)[] | Set<string>): ObjectId[] {
   return [...ids].map((id) => new ObjectId(id));

